custom_components/blitzortung/geo_location.py

```diff
@@ -139,75 +150,33 @@ class BlitzortungEvent(GeolocationEvent):
 
     def __init__(self, distance, latitude, longitude, unit, time, status, region):
         """Initialize entity with data provided."""
-        self._distance = distance
-        self._latitude = latitude
-        self._longitude = longitude
         self._time = time
         self._status = status
         self._region = region
         self._publication_date = time / 1e9
         self._remove_signal_delete = None
         self._strike_id = str(uuid.uuid4()).replace("-", "")
-        self._unit_of_measurement = unit
-        self.entity_id = "geo_location.lightning_strike_{0}".format(self._strike_id)
-
-    @property
-    def extra_state_attributes(self):
-        """Return the device state attributes."""
-        attributes = {}
-        for key, value in (
-            (ATTR_EXTERNAL_ID, self._strike_id),
-            (ATTR_ATTRIBUTION, ATTRIBUTION),
-            (ATTR_PUBLICATION_DATE, utc_from_timestamp(self._publication_date)),
-        ):
-            attributes[key] = value
-        return attributes
-
-    @property
-    def distance(self):
-        """Return distance value of this external event."""
-        return self._distance
-
-    @property
-    def icon(self):
-        """Return the icon to use in the front-end."""
-        return DEFAULT_ICON
-
-    @property
-    def latitude(self):
-        """Return latitude value of this external event."""
-        return self._latitude
-
-    @property
-    def longitude(self):
-        """Return longitude value of this external event."""
-        return self._longitude
-
-    @property
-    def name(self):
-        """Return the name of the event."""
-        return DEFAULT_EVENT_NAME_TEMPLATE.format(self._publication_date)
-
-    @property
-    def source(self) -> str:
-        """Return source value of this external event."""
-        return DOMAIN
-
-    @property
-    def should_poll(self):
-        """Disable polling."""
-        return False
-
-    @property
-    def unit_of_measurement(self):
-        """Return the unit of measurement."""
-        return self._unit_of_measurement
+        self.entity_id = f"geo_location.lightning_strike_{self._strike_id}"
+        self._attr_name = "Lightning strike"
```

Comment:
Nice, didn't know this trick with `_attr` prefix.

Reply:
Yeah this is very useful.